GPU/Common/FramebufferCommon.cpp

```diff
@@ -1998,3 +1999,13 @@ void FramebufferManagerCommon::ReadFramebufferToMemory(VirtualFramebuffer *vfb,
 	}
 }
 
+void FramebufferManagerCommon::FlushBeforeCopy() {
+	// Flush anything not yet drawn before blitting, downloading, or uploading.
+	// This might be a stalled list, or unflushed before a block transfer, etc.
+
+	// TODO: It's really bad that we are calling SetRenderFramebuffer here with
+	// all the irrelevant state checking it'll use to decide what to do. Should
+	// do something more focused here.
```

Comment:
We could probably breakpoint cases where this happens and reorder the logic.  I barely remember, but I think it's usually already set and this does nothing.

The common case should be memcpy() from the CPU, where the render buffer will most often already be set (but might not be flushed.)  Maybe this was happening in Jeanne d'Arc or something... I think it was a block transfer?

I know this is cargo culting, just drive-bying.

-[Unknown]

Reply:
Yeah, mostly this probably isn't even necessary..

Reply:
Well, it does have a check where if it's already set it does nothing.  Hopefully it hits that most commonly.

-[Unknown]